packages/worker/src/boot/startWorker.ts

```diff
@@ -3,38 +3,37 @@ import { Deflate, constants, string2buf } from '../domain/deflate'
 import type { DeflateWorkerAction, DeflateWorkerResponse } from '../types'
 
 export function startWorker() {
-  monitor(() => {
+  try {
     const streams = new Map<number, Deflate>()
-    self.addEventListener(
-      'message',
-      monitor((event: MessageEvent<DeflateWorkerAction>) => {
+    self.addEventListener('message', (event: MessageEvent<DeflateWorkerAction>) => {
+      try {
         const response = handleAction(streams, event.data)
         if (response) {
           self.postMessage(response)
         }
-      })
-    )
-  })()
+      } catch (error) {
+        sendError(error, event.data && 'streamId' in event.data ? event.data.streamId : undefined)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Could we have a test to ensure that the streamId is well sent with telemetry errors?

Reply:
Ah, I didn't understood that you meant to send the stream id to telemetry errors. In my mind the stream id was insignificant, only a number incremented each time a "DeflateWriter" was created. But we can surely hardcode a stream id for each use case, it might help debugging issue. I'll add it to telemetry errors.